src/project-tests/technical-docs-tests.js

```diff
@@ -59,7 +59,8 @@ export default function createTechnicalDocsPageTests() {
                     
                     const headerText = mainSections.map((el, i) => {
                       FCC_Global.assert.isAbove(el.firstElementChild.innerText.length, 0, 'All headers must contain text ')
-                      return [][i] = el.firstElementChild.innerText.replace(/\s/g, '_');
+                      //remove leading / trailing spaces in headerText, replace all other spaces with underscores, and replace the ellipsis with an underscore.
+                      return [][i] = el.firstElementChild.innerText.trim().replace(/^\s/g, '').replace(/\s$/g, '').replace(/\s/g, '_').replace('...', '_');
                     });
```

Comment:
@tbushman @tchaffee so I don't think we need both `.trim()` and the regex here. Either or will work, but trim should be sufficient. If we were to use regex, we could simplify it into one regex instead of chaining 2 `.replace()`s for the trailing and leading spaces, and make sure it covers more than 1 space just to be safe:
```js
'   example   '.replace(/^\s+|\s+$/g, '');
'   example    '.trim();
```

both of these will yield the same result. So this is something that can and should be cleaned up. 

The same goes for the elipsis and the `\s` character. We can simplify this instead of chaining by using the regex OR operator `|`. 

The best solution here I think is:
```js
'   examples...are cool   '.trim().replace(/\s|\.\.\./g, '_'); //  => examples_are_cool
```
Try running this line in the browser console or on https://repl.it/languages/javascript and you'll see what I mean. 

It has been a while since I've touched this project so if there was another reason for doing it like this I'm all ears! Otherwise, I think this can be changed to reflect the above.